### PR TITLE
Throw if `cwd` option isn't a `string`

### PR DESCRIPTION
**Pull request by dead-horse**
[globby](https://github.com/sindresorhus/globby) will pass `opts.cwd = undefined` to dir-glob, which leads to TypeError.

### CODE REVIEW

**Comment by dead-horse** [comment]
ping @sindresorhus

**Comment by sindresorhus** [comment]
I think this should be fixed in `globby`. The option handling here is correct.

**Comment by dead-horse** [comment]
`options.cwd` could be undefined or null somehow, I think we should ensure dir-glob won't throw here?

**Comment by sindresorhus** [comment]
IMHO, it's up to the caller to pass in the correct arguments. `dir-glob` should not handle invalid input.

**Comment by dead-horse** [comment]
at least we should alert user `opts.cwd` set to undefined is invalid to avoid this `TypeError` which is hard to debug.

**Comment by sindresorhus** [comment]
Agreed

**Comment by dead-horse** [comment]
you'll fix this in `globby` and I change this PR to alert this invalid input?

**Comment by doochik** [comment]
I've created PR to globby https://github.com/sindresorhus/globby/pull/99

**Comment by hashar** [comment]
See also:

https://github.com/kevva/dir-glob/issues/10
https://github.com/sindresorhus/globby/issues/98

:)

File: index.js [hunk 1]
```diff
@@ -24,9 +24,11 @@ const getGlob = (dir, opts) => {
 
 	if (opts.files && opts.extensions) {
 		return opts.files.map(x => path.join(dir, addExtensions(x, opts.extensions)));
-	} else if (opts.files) {
+	}
+	if (opts.files) {
```
Comment (kevva) [inline index.js:29]: These changes are unrelated.

File: package.json [hunk 2]
```diff
@@ -29,10 +29,10 @@
 		"path-type": "^3.0.0"
 	},
 	"devDependencies": {
-		"ava": "*",
+		"ava": "~0.25.0",
```
Comment (kevva) [inline package.json]: Also unrelated.

File: index.js [hunk 3]
```diff
@@ -43,6 +48,9 @@ module.exports = (input, opts) => {
 
 module.exports.sync = (input, opts) => {
 	opts = Object.assign({cwd: process.cwd()}, opts);
+	if (typeof opts.cwd !== 'string') {
+		throw new TypeError('opts.cwd must be a string');
```
Comment (kevva) [inline index.js]:
```js
new TypeError(`Expected `cwd` to be of type \`string\` but received type \`${typeof opts.cwd}\``);
```

**Review by kevva — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (dead-horse) [reply index.js]: error message updated, all the other changes made the project pass the lint by xo, pin `ava` and `xo` to ensure it work in node v4.

Comment (dead-horse) [reply index.js]: I've make these changes into two commit if you want to separate them.

**Review by kevva — APPROVED** [review]

**Comment by kevva** [comment]
Thanks @dead-horse!
